### PR TITLE
Add null storage, use for tangle benchmarking

### PR DESCRIPTION
**Pull request by Adam-Gleave**
# Description of change

Adds a storage layer that doesn't interact with any underlying data and just satisfies the requirements of the `StorageBackend` trait, in order to use the tangle for testing purposes without requiring a database.

Also adds benchmarking for `insert`/`update_metadata` using this storage.

Removes the need for #765.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ members = [
 	"bee-storage/bee-storage-rocksdb",
 	"bee-storage/bee-storage-sled",
 	"bee-storage/bee-storage-test",
+	"bee-storage/bee-storage-null",
```
Comment (thibault-martinez) [inline Cargo.toml]: order!

File: bee-storage/bee-storage-null/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+### Fixed
+
+### Security -->
+
+## 0.1.0 - 2021-10-11 
+
+### Added
+
+- Implementation of `StorageBackend` for `NullStorage`;
```
Comment (thibault-martinez) [inline bee-storage/bee-storage-null/CHANGELOG.md]:
```suggestion
- Implementation of `StorageBackend` for `Storage`;
```
Type is just called Storage.

File: bee-storage/bee-storage-null/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "bee-storage-null"
+version = "0.1.0"
+authors = ["IOTA Stiftung"]
+edition = "2018"
+description = "A bee-storage implementation with no backend, for testing purposes"
+readme = "README.md"
+repository = "https://github.com/iotaledger/bee"
+license = "Apache-2.0"
+keywords = ["iota", "tangle", "bee", "framework"]
```
Comment (thibault-martinez) [inline bee-storage/bee-storage-null/Cargo.toml]: Add storage keyword

File: bee-storage/bee-storage-null/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    fn insert(&self, _key: &T, _value: &U) -> Result<(), Self::Error> {
+        Ok(())
+    }
+}
+
+impl<T, U> Fetch<T, U> for Storage {
+    fn fetch(&self, _key: &T) -> Result<Option<U>, Self::Error> {
+        Ok(None)
+    }
+}
```
Comment (thibault-martinez) [inline bee-storage/bee-storage-null/src/lib.rs]: Can you implement all access operations ?

File: bee-tangle/Cargo.toml [hunk 5]
```diff
@@ -27,5 +27,16 @@ rand = "0.8"
 ref-cast = "1.0"
 serde = { version = "1.0", features = [ "derive" ] }
 thiserror = "1.0"
-tokio = { version = "1.11", features = [ "sync", "time" ] }
+tokio = { version = "1.11", features = [ "rt", "rt-multi-thread", "sync", "time" ] }
```
Comment (thibault-martinez) [inline bee-tangle/Cargo.toml]: Maybe put tokio with these features only in dev-deps ?

Comment (thibault-martinez) [inline bee-tangle/Cargo.toml]: Maybe put tokio with these features only in dev-deps ?

File: bee-tangle/Cargo.toml [hunk 6]
```diff
@@ -27,5 +27,16 @@ rand = "0.8"
...
 tokio-stream = { version = "0.1" }
+
+[dev-dependencies]
+bee-storage-null = { version = "0.1.0", path = "../bee-storage/bee-storage-null" }
+bee-test = { version = "0.1.0", path = "../bee-test" }
+
+criterion = { version = "0.3.5", features = [ "async_tokio" ] }
```
Comment (thibault-martinez) [inline bee-tangle/Cargo.toml]:
```suggestion
criterion = { version = "0.3", features = [ "async_tokio" ] }
```

**Review by Alex6323 — DISMISSED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (Alex6323) [reply Cargo.toml]: :de:

Comment (Adam-Gleave) [reply Cargo.toml]: 🔨

File: bee-storage/bee-storage-null/src/lib.rs (hunk 4 above)
Comment (Adam-Gleave) [reply bee-storage/bee-storage-null/src/lib.rs]: Yes, will do.

**Review by thibault-martinez — COMMENTED** [review]
I know it's very annoying but now it's triggering me a tiny bit that some access ops have their module and some don't. Align !?

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**